### ISSUE TITLE
CodeEditor "readonly" style allows adding newline characters to UI

### ISSUE THREAD

**Issue by domspad** [enthought/traitsui#264]
can be seen in the current `traitsui/examples/demo/demo.py` file or in the following minimal example

```
from traits.api import (
    HasStrictTraits, Code
)
from traitsui.api import View, Item, CodeEditor


class CodeView(HasStrictTraits):
    """ A read-only view of the raw data that is being loaded in the data_frame

    """
    code = Code

    def _code_default(self):
        return '#No code'

    traits_view = View(
        Item(
            'code', show_label=False, springy=True, full_size=True,
            resizable=True, style='readonly', editor=CodeEditor()
        ),
        resizable=True,
    )

cv = CodeView()

cv.configure_traits()
```

**Comment by corranwebster** [comment]
Removed reference to private repo. @domspad please be careful about this - it was an internal project not a client project, so not terribly serious, but this would have been a serious matter for some of out clients.  The way to link is to reference this ticket from the internal repo (which I will do shortly).

All that aside, I assume this is observed under Qt?

**Comment by corranwebster** [comment]
It's possible the bug here is in the underlying Pyface python console widget, so that may end up being where the issue needs to be raised and the fix applied.

**Comment by domspad** [comment]
@corranwebster sorry about that! Thanks for catching it so quickly. I should know better than to reference so haphazardly.

Anyway, yes this was all Qt. I spent a good bit trying to follow my way down the traitsui code, but had to move on. The funny thing is both `CodeEditor` and `TextEditor` are made with `EditorFactory`, and are very similar editors, but only `TextEditor` has a `read_only` trait, while `CodeEditor` instead relies on (apparently), the `style` trait of the `Item` it is used by. `TextEditor` does not exhibit the problem mentioned here, just `CodeEditor`, so I tried, as on possible fix, adding a `read_only` trait to `CodeEditor`, but to no avail.
